src/rules/selector-no-redundant-nesting-selector/index.js

```diff
@@ -15,18 +15,43 @@ export default function(actual) {
     }
 
     root.walkRules(/&/, rule => {
-      const { selector } = rule;
-      // "Ampersand followed by a combinator followed by non-combinator non-ampersand and not the selector end"
-      const regex = /^&\s*[>+~ ]\s*[^>+~ {&]+/;
-
-      if (selector === "&" || regex.test(selector)) {
-        utils.report({
-          ruleName,
-          result,
-          node: rule,
-          message: messages.rejected
+      parseSelector(rule.selector, result, rule, fullSelector => {
+        // "Ampersand followed by a combinator followed by non-combinator non-ampersand and not the selector end"
+        fullSelector.walkNesting(node => {
+          const prev = node.prev();
+
+          if (prev) {
+            return;
+          }
+
+          const next = node.next();
+
+          if (!next && node.parent.parent.nodes.length > 1) {
```

Comment:
@kristerkari any nested element have `parent` and any `parent` have `parent` as full selector, but in next version behavior in theory can be changed, we can add more checks to avoid this situation.